src/scanners/javascript.js

```diff
@@ -13,6 +14,12 @@ export default class JavaScriptScanner {
   constructor(code, filename) {
     this.code = code;
     this.filename = filename;
+
+    if (this.filename === undefined || this.filename === null ||
```

Comment:
Good point, and yeah I forgot about typeof, oops!

I'll clean it up a bit. Testing for a string makes sense too... Maybe it should just be typeof === string and length > 0

- tofumatt

> On 11 Nov 2015, at 10:04, Stuart Colville <notifications@github.com> wrote:
> 
> In src/scanners/javascript.js:
> 
> > @@ -13,6 +14,12 @@ export default class JavaScriptScanner {
> >    constructor(code, filename) {
> >      this.code = code;
> >      this.filename = filename;
> > +
> > +    if (this.filename === undefined || this.filename === null ||
> Seeing this twice makes me think a util func might be worth doing.
> 
> —
> Reply to this email directly or view it on GitHub.
>